datadogclient/datadog_client.go

```diff
@@ -129,32 +150,80 @@ func (c *Client) AddMetric(envelope *events.Envelope) {
 	c.metricPoints[key] = mVal
 }
 
-func (c *Client) PostMetrics() error {
+func (c *Client) addEvent(envelope *events.Envelope) {
+	title := fmt.Sprintf("%s: %d", envelope.GetError().GetSource(), envelope.GetError().GetCode())
+	event := Event{
+		Title:        title,
+		Text:         envelope.GetError().GetMessage(),
+		DateHappened: envelope.GetTimestamp() / int64(time.Second),
+		Tags:         parseTags(envelope),
+	}
+
+	c.events = append(c.events, event)
+}
+
+func (c *Client) Post() error {
 	c.populateInternalMetrics()
+
+	if err := c.postMetrics(); err != nil {
+		return err
+	}
+
+	if err := c.postEvents(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) postMetrics() error {
 	numMetrics := len(c.metricPoints)
 	c.log.Infof("Posting %d metrics", numMetrics)
 
 	c.totalMetricsSent += uint64(len(c.metricPoints))
-	seriesBytes := c.formatter.Format(c.prefix, c.maxPostBytes, c.metricPoints)
+	seriesBytes := c.formatter.FormatMetrics(c.prefix, c.maxPostBytes, c.metricPoints)
 	c.metricPoints = make(map[MetricKey]MetricValue)
 
-	for _, data := range seriesBytes {
-		if uint32(len(data)) > c.maxPostBytes {
-			c.log.Infof("Throwing out metric that exceeds %d bytes", c.maxPostBytes)
+	return c.sendData(seriesBytes, c.seriesURL())
+}
+
+func (c *Client) postEvents() error {
+	numEvents := len(c.events)
+	c.log.Infof("Posting %d events", numEvents)
+
+	c.totalMetricsSent += uint64(numEvents)
```

Comment:
Can you use a different tracker for this? We usually track these differently in most cases. So you'd need to change the log lines and add this one in there as well, but I think it will be more useful 😄